### PR TITLE
Check and locate introduction of NA's

### LINKED ISSUE #13: apply_data_dictionary(): Print messages if NA's are introduces or removed
Using `apply_data_dictionary()` can coerce column types and can change the levels of factor variables.

This can lead to the introduction or removal of `NA` values.

While introducing/removing `NA`'s might be the desired behaviour, it can be the source for a lot of confusion and frustration in further analysis.
Therefore, it would be useful if `apply_data_dictionary()` would message if and for which variables NA's were introduced or removed.

**Suggestion**: Add a function that is called by `apply_data_dictionary()` that checks whether `NA`'s were introduced/removed and message where this happened. Make this a message to reflect that it can be intentional to remove/introduce `NA`'s

### LINKED ISSUE #19: Turn .find_NA_coercions() into a standalon function
The idea behind our data import suite is:

- `apply_data_dictionary()` is the primary entry point for the user. Essentially, it is a wrapper for more specialised functions working with a `data_dictionary`

- If the user wants to access the functions that are called by `apply_data_dictionary()` the user can.

`.find_NA_coercions()` can already be used as a standalone function but currently relies on a "data_dictionary" matching old and new columns names. To make `.find_NA_coercions()` easily useable as an external function I suggest adding an optional argument in which matching columns can be specified.

The function could then be used either by providing a data_dictionary or by "manually"  specifying old and new columns.

### PR DESCRIPTION
**Pull request by Besttd**
Added a helper function that checks if and locates where NA's are introduced in the process of calling `apply_data_dictionary()`.

This is called from within `apply_data_dictionary()` and prints all values that were changed to NA's.
If NA introduction is intentional, this behaviour can be turned off by setting `apply_data_dictionary(print_coerced_NA = FALSE)`

Closes #13

### CODE REVIEW

File: R/Tidy_names_types_factors.R [hunk 1]
```diff
@@ -35,10 +38,14 @@
 #' @export
 #'
 #' @author J. Peter Marquardt
-apply_data_dictionary <- function(data, data_dictionary) {
+apply_data_dictionary <- function(data, data_dictionary, print_coerced_NA = TRUE) {
 
   assertive.types::is_data.frame(data)
   assertive.types::is_data.frame(data_dictionary)
+  assertive.types::is_a_bool(print_coerced_NA)
```
Comment (p-mq) [inline R/Tidy_names_types_factors.R:58]: Need to update assertive.types dependencies

File: R/Tidy_names_types_factors.R [hunk 2]
```diff
@@ -267,13 +284,13 @@ parse_date_columns <- function(data, date_formats) {
 
 #' Parse a string to create a named list
 #'
-#' Create a named list from a standardised string of the follwoing format:
+#' Create a named list from a standardized string of the following format:
```
Comment (p-mq) [inline R/Tidy_names_types_factors.R]: Are you suggestion an AE only style convention?

File: R/Tidy_names_types_factors.R [hunk 3]
```diff
@@ -307,3 +324,43 @@ parse_date_columns <- function(data, date_formats) {
 
   return(vct)
 }
+
+
+#' Locate NA values introduced during data cleaning
```
Comment (p-mq) [inline R/Tidy_names_types_factors.R]: Consider replacing "data cleaning" with "`apply_data_dictionary()`" to be more verbose on non-general use

File: R/Tidy_names_types_factors.R [hunk 4]
```diff
@@ -307,3 +324,43 @@ parse_date_columns <- function(data, date_formats) {
...
+#' @keywords internal
+#'
+#' @author Till D. Best, J. Peter Marquardt
+.find_NA_coercions <- function(data_raw, data, data_dictionary) {
+  #  create a dataframe matchin the old and new column name in our data dictionary
+  rosetta_stone <- data.frame(old_name = data_dictionary$old_column_name[!is.na(data_dictionary$new_data_type)],
+                              new_name = colnames(data))
+
+  # find those value where NA's were introduced
+  NA_difference <- is.na(data_raw[rosetta_stone$old_name]) < is.na(data)
```
Comment (p-mq) [inline R/Tidy_names_types_factors.R:396]: If we made this a parameter, could we use this method in a more general application? That might actually be worth considering

**Review by p-mq — CHANGES_REQUESTED** [review]
Looking great already!

If you don't want to upgrade the function to work in the more general case, I suggest adding it as an issue for version 2.0

File: R/Tidy_names_types_factors.R (hunk 1 above)
Comment (Besttd) [reply R/Tidy_names_types_factors.R:58]: fixed

File: R/Tidy_names_types_factors.R (hunk 2 above)
Comment (Besttd) [reply R/Tidy_names_types_factors.R]: reverted

File: R/Tidy_names_types_factors.R (hunk 4 above)
Comment (Besttd) [reply R/Tidy_names_types_factors.R:396]: see my general comment

File: R/Tidy_names_types_factors.R (hunk 3 above)
Comment (Besttd) [reply R/Tidy_names_types_factors.R]: done

**Review by Besttd — COMMENTED** [review]
As discussed in person:
- .find_NA_coercions can already be used as a standalone which is not reflected in the documentation at the moment.
- maturing `.find_NA_coercions()` to a standalone function should be a goal for 2.0.

**Review by p-mq — APPROVED** [review]
Thanks, good work!

**Review by p-mq — APPROVED** [review]

**Pull request merged**